### PR TITLE
Update find command to perform and search

### LINKED ISSUE #232: Update find command to be AND search
OR search performed when specify both n/ and d/ feels weird

### LINKED ISSUE #246: Update find command to be AND search

### PR DESCRIPTION
**Pull request by Feng1231**
Find command finds students whose name contains one of the name keywords provided *OR* their schedule is within one of the days provided. This does not feel intuitive to the user.

Let's update find command to find students whose name contains one of the name keywords provided *AND* their schedule is within one of the days provided.

closes #246

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 1]
```diff
@@ -116,8 +128,8 @@ public void execute_multipleScheduleKeywords_multipleStudentsFound() {
     }
 
     @Test
-    public void execute_nameAndScheduleKeywords_multipleStudentsFound() {
-        String expectedMessage = String.format(MESSAGE_STUDENTS_LISTED_OVERVIEW, 2);
+    public void execute_nameAndScheduleKeywords_noStudentFound() {
+        String expectedMessage = String.format(MESSAGE_STUDENTS_LISTED_OVERVIEW, 0);
```
Comment (DennieDan) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:132]: Looks good!
But if so, I think need to specify in the UG also, e.g., give an example to tell them that if both name and day are given, then only search for students containing 1 of the names **and** 1 of the days.

**Review by DennieDan — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java (hunk 1 above)
Comment (Feng1231) [reply src/test/java/seedu/address/logic/commands/FindCommandTest.java:132]: Hi, thank you for catching this. documentation updated here. #254

**Review by DennieDan — APPROVED** [review]
LGTM! Thanks for changing

**Review by TY1Fan — APPROVED** [review]
LGTM

**Pull request merged**